apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -374,12 +381,25 @@ const Ticket = new GQLListSchema('Ticket', {
 
             if (userType === RESIDENT && operation === 'create') {
                 overrideTicketFieldsForResidentUserType(context, resolvedData)
-                const contact = await getOrCreateContact(context, resolvedData)
-                resolvedData.contact = contact.id
                 await setSectionAndFloorFieldsByDataFromPropertyMap(context, resolvedData)
                 setClientNamePhoneEmailFieldsByDataFromUser(get(context, ['req', 'user']), resolvedData)
             }
 
+            if (!resolvedContact) {
```

Comment:
Could you please extract all this logic inside helper function like `setSectionAndFloorFieldsByDataFromPropertyMap`